map more Reviewer information (PUBD-985)

hardyoyo opened the pull request:
- Add the post_process method that @everreau suggested in PUBD-985

hardyoyo committed:
WIP, sketch out the post_process method that @everreau suggested in PUBD-985

hardyoyo commented on serializers.py:
```diff
@@ -384,6 +384,17 @@ def create(self, validated_data: dict) -> Account:
         except Account.DoesNotExist:
             return super().create(validated_data)
+        
+    #TODO add a post_process method to create interest objects for the user and
```
Remove TODO comment

hardyoyo committed:
remove TODO that is done

hardyoyo commented:
Currently deployed to dev, ready for testing

hardyoyo commented:
@justingonder recommends this as a test procedure:

> wipe and re-transport of CPCEM on dev

hardyoyo commented:
docs on how this is all supposed to work: https://github.com/cdlib/pad-sys-doc/wiki/OJS-to-Janeway-(AKA-Journal-Transporter)

hardyoyo committed:
PUBD-985: Add user.save() to post_process for interests

hardyoyo committed:
Refactor to add Interest objects, then associate them with the reviewer

bibliotechy commented on serializers.py:
```diff
@@ -381,9 +386,36 @@ def create(self, validated_data: dict) -> Account:
+                # first add a new interest object
+                # if it already exists, it will just return the existing one
+                # if it doesn't exist, it will create a new one
+                interest_object = Interest.objects.get_or_create(name=interest)
```
variable unpacking might make this a little more readable.
`interest_object, was_created = Interest.objects.get_or_create(name=interest)`

bibliotechy commented on serializers.py:
```diff
@@ -381,9 +386,36 @@ def create(self, validated_data: dict) -> Account:
+                # if it doesn't exist, it will create a new one
+                interest_object = Interest.objects.get_or_create(name=interest)
+                # next, save the interest object, just in case
+                interest_object[0].save()
```
I think we only need to save it if it was created. If it wasn't created, then it already exists and isn't being updated, right?

hardyoyo committed:
PUBD-985: skip invalid users

hardyoyo committed:
- Create new add_interests_to_user method on TransporterSerializer
- Call this method in UserSerializer's post_process method
- Call UserSerializer's post_process method in the JournalArticleAuthorSerializer child class, which derives from UserSerializer

hardyoyo commented:
![image](https://github.com/eScholarship/journal-transporter-janeway-1.4-plugin/assets/611569/b8990084-2d0e-4f10-9dea-dc4f43485d42)

hardyoyo commented:
![reviewer-interests](https://github.com/eScholarship/journal-transporter-janeway-1.4-plugin/assets/611569/955c433d-c385-451d-98cd-1cec3696293a)

hardyoyo commented:
Two successful journal migration runs for journals with interests that need transporting: uclastat_cts_ties and uciem_cpcem, this code is ready for review, approval and merge

bibliotechy commented on serializers.py:
```diff
@@ -260,6 +258,23 @@ def handle_attachments(self, object) -> None:
+                    # first add a new interest object
+                    # if it already exists, it will just return the existing one
+                    # if it doesn't exist, it will create a new one
+                    interest_object = Interest.objects.get_or_create(name=interest)
```
I think the idiomatic way to write this is with variable expansion, as in 
```python
interest_object, was_created = Interest.objects.get_or_create(name=interest)
```
That way `interest_object[0]` becomes `interest_object` in all the places you need to use it.

bibliotechy reviewed (approved):
One stylistic comment that you can take or leave.

hardyoyo committed:
cleanup suggested by @bibliotechy
